src/hip_module.cpp

```diff
@@ -546,6 +547,12 @@ hipError_t ihipModuleLoadData(hipModule_t* module, const void* image) {
     auto ctx = ihipGetTlsDefaultCtx();
     if (!ctx) return hipErrorInvalidContext;
 
+    // try extracting code object from image as fatbin.
+    char name[64] = {};
+    hsa_agent_get_info(this_agent(), HSA_AGENT_INFO_NAME, name);
+    if (auto *code_obj = __hipExtractCodeObjectFromFatBinary(image, name))
```

Comment:
Alternatively if we still need the above abort, we can call this piece of code only when hip-clang is used as the compiler.

Reply:
As commented above, apps compiled by hcc will try to load a code object, therefore will not abort. Abort only happens when apps compiled by hip-clang tries to load a corrupted fat binary.

Reply:
The host code could be compiled by gcc, therefore there is no way to know the fat bin is intended for HCC or hip-clang. I have removed abort from code above so that fat bin for HCC can be handled in other places.